src/pgduckdb_ddl.cpp

```diff
@@ -392,12 +393,34 @@ DuckdbHandleDDL(PlannedStmt *pstmt, const char *query_string, ParamListInfo para
 	} else if (IsA(parsetree, CreateForeignServerStmt)) {
 		DuckdbHandleCreateForeignServerStmt(parsetree);
 		return;
+	} else if (IsA(parsetree, AlterForeignServerStmt)) {
+		DuckdbHandleAlterForeignServerStmt(parsetree);
+		return;
 	} else if (IsA(parsetree, CreateUserMappingStmt)) {
 		DuckdbHandleCreateUserMappingStmt(parsetree);
 		return;
 	}
 }
 
+static void DuckdbHandleAlterForeignServerStmt(Node *parsetree) {
+	pgduckdb::CurrentServerType = nullptr;
+	// Propagate the "TYPE" to the server options, don't validate it here though
+	auto stmt = castNode(AlterForeignServerStmt, parsetree);
+	auto server = GetForeignServerByName(stmt->servername, false);
+
+	// XXX - should we cache this?
```

Comment:
@JelteF would love your opinion on this one.

Reply:
Nah, `get_foreign_data_wrapper_oid` uses `GetSysCacheOid1` internally. So it's already using a cache. The main reason we have our own cache for some things is so that we can access it in a thread-safe way, which we don't need here.

Even without all that, this isn't a hot code path. So caching in general seems like a waste.